core/org.openjdk.jmc.flightrecorder.writer/src/main/java/org/openjdk/jmc/flightrecorder/writer/TypedFieldImpl.java

```diff
@@ -49,7 +49,7 @@
 	private final boolean isArray;
 	private final List<Annotation> annotations;
 
-	TypedFieldImpl(TypeImpl type, String name) {
+	public TypedFieldImpl(TypeImpl type, String name) {
```

Comment:
Now that the implementation classes are becoming public, do you wish to move them off to a separate package to signal that people shouldn't rely on them, and that they are not part of the public API?

Reply:
The API classes are in `org.openjdk.jmc.flightrecorder.writer.api` package which is the only one exported.
However, if you feel strongly about this I will move the impl classes to `org.openjdk.jmc.flightrecorder.writer.impl` package instead.

Reply:
Personally I think I might have gone with api being in the writer top-level package, and then an impl package, but it's probably not worth the heartache to change that. So, nah don't feel strongly about it.

Reply:
Ok. You are right, it would have been better in the top package. IDK, I just turned on auto pilot and did the `api` package - I used to put various modular/infra classes into the module package. Probably this approach is not that good any more.

But given that the API has been already made public we will have to live with that :(